extensions/datastores/kudu/src/main/java/org/locationtech/geowave/datastore/kudu/operations/BatchHandler.java

```diff
@@ -0,0 +1,36 @@
+package org.locationtech.geowave.datastore.kudu.operations;
+
+import org.apache.kudu.client.KuduSession;
+import org.apache.kudu.client.SessionConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class BatchHandler {
+  protected final KuduSession session;
+  private static final Logger LOOGER = LoggerFactory.getLogger(BatchHandler.class);
+
+  public BatchHandler(final KuduSession session) {
+    this.session = session;
+  }
+
+  protected boolean setAutoFlushMode() {
+    session.setFlushMode(SessionConfiguration.FlushMode.AUTO_FLUSH_BACKGROUND);
+    if (session.getFlushMode() != SessionConfiguration.FlushMode.AUTO_FLUSH_BACKGROUND) {
+        LOOGER.error("Fail to set session Flush Mode to AUTO_FLUSH_BACKGROUND.");
+        return false;
+    }
+    return true;
+  }
+
+  protected boolean setDefaultFlushMode() {
```

Comment:
I'm a little confused by this. If the default flushing mode is synchronous, doesn't that mean it's already batched? My understanding is that if we are autoflushing, then the operations are not necessarily batched.